test/plots/segmentPlotTests.ts

```diff
@@ -2,159 +2,168 @@
-      plot.x(function(d) { return d.x; }, xScale);
-      plot.y(function(d) { return d.y; }, yScale);
-      plot.y2(function(d) { return d.y2; });
-      plot.addDataset(new Plottable.Dataset(data)).renderTo(svg);
-      renderArea = (<any> plot)._renderArea;
-      renderArea.selectAll("line")[0].forEach((line) => {
-      let lineSelection = d3.select(line);
-      assert.strictEqual(lineSelection.attr("x1"), lineSelection.attr("x2"), "line is vertical");
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(500, 500);
+        xScale = new Plottable.Scales.Linear();
+        yScale = new Plottable.Scales.Linear();
       });
-      svg.remove();
-    });
 
-    it("renders horizontal lines when y2 is not set", () => {
-      let plot = new Plottable.Plots.Segment();
-      plot.x(function(d) { return d.x; }, xScale);
-      plot.x2(function(d) { return d.x2; });
-      plot.y(function(d) { return d.y; }, yScale);
-      plot.addDataset(new Plottable.Dataset(data)).renderTo(svg);
-      renderArea = (<any> plot)._renderArea;
-      renderArea.selectAll("line")[0].forEach((line) => {
-      let lineSelection = d3.select(line);
-      assert.strictEqual(lineSelection.attr("y1"), lineSelection.attr("y2"), "line is horizontal");
+      afterEach(() => {
+        svg.remove();
       });
-      svg.remove();
-    });
 
-    it("autorangeMode(\"x\")", () => {
-      let staggeredData = [
-        { y: 0, x: 0, x2: 1 },
-        { y: 1, x: 1, x2: 2 }
-      ];
-      xScale.padProportion(0);
+      it("renders a line properly", () => {
+        let plot = new Plottable.Plots.Segment();
+        plot.x((d) => d.x, xScale);
+        plot.x2((d) => d.x2);
+        plot.y((d) => d.y, yScale);
+        plot.y2((d) => d.y2);
+        plot.addDataset(new Plottable.Dataset([data[0]]));
+        plot.renderTo(svg);
+
+        let line = plot.content().selectAll("line");
+        assert.strictEqual(line.size(), 1, "exactly one line has been rendered");
+        assert.strictEqual(+line.attr("x1"), 62.5, "x1 is correct");
+        assert.strictEqual(+line.attr("x2"), 437.5, "x2 is correct");
+        assert.strictEqual(+line.attr("y1"), 437.5, "y1 is correct");
+        assert.strictEqual(+line.attr("y2"), 62.5, "y2 is correct");
+      });
 
-      let plot = new Plottable.Plots.Segment();
-      plot.x(function(d) { return d.x; }, xScale);
-      plot.x2(function(d) { return d.x2; });
-      plot.y(function(d) { return d.y; }, yScale);
-      plot.addDataset(new Plottable.Dataset(staggeredData));
-      plot.autorangeMode("x");
-      plot.renderTo(svg);
+      it("renders vertical lines when x2 is not set", () => {
+        let plot = new Plottable.Plots.Segment();
+        plot.x((d) => d.x, xScale);
+        plot.y((d) => d.y, yScale);
+        plot.y2((d) => d.y2);
+        plot.addDataset(new Plottable.Dataset(data));
+        plot.renderTo(svg);
+
+        plot.content().selectAll("line")[0].forEach((line) => {
```

Comment:
The "more d3" way to iterate over would be to do:
```Typescript
plot.content().selectAll("line").each(function() {
  let lineSelection = d3.select(this); // "this" refers to the DOM node inside the loop
  ...
});
```
However, I think the syntax you used is clearer. We should standardize around one or the other.